frontend/src/app/main/ui/workspace/sidebar/assets/components.cljs

```diff
@@ -193,7 +207,7 @@
 
         components     (not-empty (get groups "" []))
         can-combine?   (and is-local
-                            (> (count components) 1)
+                            (> (count-leaves groups) 1)
```

Comment:
This operation were constant time, now it is O(n). I suggest putting it under `mf/with-memo`